src/esbmc/esbmc_parseoptions.cpp

```diff
@@ -1684,6 +1685,22 @@ bool esbmc_parseoptionst::process_goto_program(
       return true;
     }
 
+    if(cmdline.isset("output-goto"))
+    {
+      msg.status("Writing GOTO program to file");
+      std::string fullname = cmdline.args[0];
+      // ! assume correct filename
+      size_t lastindex = fullname.find_last_of(".");
+      std::string gbfname = fullname.substr(0, lastindex) + ".goto";
```

Comment:
Please don't modify the path the user passed as argument: you might be overwriting important data unrelated to this run of ESBMC. Generally speaking, making assumptions about filename extensions is not portable.